packages/hooks/src/data/QueryData.ts

```diff
@@ -224,13 +218,21 @@ export class QueryData<TData, TVariables> extends OperationData {
       this.currentObservable.query = this.refreshClient().client.watchQuery(
         observableQueryOptions
       );
+
+      if (this.context && this.context.renderPromises) {
+        this.context.renderPromises.registerSSRObservable(
+          this.currentObservable.query,
+          observableQueryOptions
+        );
+      }
     }
   }
 
   private updateObservableQuery() {
     // If we skipped initially, we may not have yet created the observable
     if (!this.currentObservable.query) {
       this.initializeObservableQuery();
+      return;
```

Comment:
this was added as an optimization. updating the options doesn't need to happen because it was just initialized.